src/apps/chifra/pkg/cache/cache.go

```diff
@@ -28,17 +29,32 @@ func getCacheAndChainPath(chain string) string {
 func save(chain string, filePath string, content io.Reader) (err error) {
 	cacheDir := getCacheAndChainPath(chain)
 	fullPath := path.Join(cacheDir, filePath)
-	file, err := os.Create(fullPath)
-	if err != nil {
-		return
+
+	var file *os.File
+	if _, err = os.Stat(fullPath); os.IsNotExist(err) {
+		// If file doesn't exist, we don't need a lock
+		file, err = os.Create(fullPath)
+		if err != nil {
+			return
+		}
+		defer file.Close()
```

Comment:
This is just a question for my own education. I C++ this line would execute at the end of this code block (if c++ had `defer`), but in GoLang, `defer` is executed at the end of the function call. Is that correct? Is that correct in every case?

Reply:
Yes, this is correct for every case.